packages/marshaller/src/ddl2/grid.ts

```diff
@@ -0,0 +1,285 @@
+import { IMonitorHandle, Widget } from "@hpcc-js/common";
+import { Cell, Grid } from "@hpcc-js/layout";
+import { IClosable, WidgetAdapter } from "@hpcc-js/phosphor";
+import { compare } from "@hpcc-js/util";
+import { IDockPanel } from "./dashboard";
+import { Element, ElementContainer } from "./model/element";
+import { IVizPopupPanelOwner, VizChartPanel, VizPopupPanel } from "./model/vizChartPanel";
+
+export class DashboardGrid extends Grid implements IClosable, IVizPopupPanelOwner, IDockPanel {
+    private _layoutCache = [];
+    constructor(private _ec: ElementContainer) {
+        super();
+
+        // this.fitTo_default("width");
+        this.designMode_default(false);
+        this.adaptiveSnapping_default(true);
+        this.showLanes_default(false);
+        this.surfacePadding_default("0");
+        this.surfaceBorderWidth_default(0);
+        // this.snappingColumns_default(900);
+        // this.snappingRows_default(900);
+        // this.squareDesignMode_default(true);
+    }
+
+    private _popups: VizPopupPanel[] = [];
+    private _popupIdx: {
+        [id: string]: {
+            panel: VizPopupPanel,
+            monitorHandle: IMonitorHandle
+        }
+    } = {};
```

Comment:
Should probably become a "PopupManager" class so that code can be commoned up between the two dashboard types?

Reply:
Can we open an issue for this and handle it separately?

Reply:
Sure